test/msd_test.py

```diff
@@ -373,6 +375,56 @@ def test_mass_storage(workspace, parent_test):
     #        when a binary file is loaded, since there is no way to
     #        tell where the end of the file is.
 
+    if page_erase_supported:
```

Comment:
I think it would be good to extract the page erase tests to another function, to keep the `test_mass_storage()` function from growing too large.

Reply:
This PR is getting big. Let me improve this in another PR because I need to another negative test to override the blocksize to be smaller to deliberately make the test fail.

Reply:
That's fine.